plugins/postcss-is-pseudo-class/src/split-selectors/compound-selector-order.ts

```diff
@@ -61,8 +61,27 @@ export function sortCompoundSelectorsInsideComplexSelector(node) {
 	}
 
 	for (let i = sortedCompoundSelectors.length - 1; i >= 0; i--) {
+		const previous = sortedCompoundSelectors[i - 1];
+
 		sortedCompoundSelectors[i].remove();
-		node.prepend(sortedCompoundSelectors[i]);
+
+		if (previous && previous.type === 'tag' && sortedCompoundSelectors[i].type === 'tag') {
+			const wrapped = parser.pseudo({
+				value: ':is',
+				nodes: [
+					parser.selector({
+						value: '',
+						nodes: [
+							sortedCompoundSelectors[i],
+						],
+					}),
+				],
+			});
+
+			node.prepend(wrapped);
+		} else {
+			node.prepend(sortedCompoundSelectors[i]);
+		}
```

Comment:
Sorting will group tag selectors `a` and `b` into `ab` which serializes as a single tag `ab`.

This bit is a "post fix" to wrap it back with `:is()` into `a:is(b)`